crates/polars-core/src/chunked_array/ops/lambda.rs

```diff
@@ -274,6 +288,17 @@ impl LambdaExpression {
                     AnyValue::Boolean(false)
                 }
             },
+            LambdaExpression::EqualNullSafe(left, right) => {
+                let left = left.eval_array(args);
+                let right = right.eval_array(args);
+                if left.is_null() && right.is_null() {
+                    AnyValue::Boolean(true)
+                } else if left.is_null() || right.is_null() {
```

Comment:
Is this redundant here? A comparison would already verify equality, as null is simply a variant
This can all just be left.eq(&right).into()

Reply:
not sure if eq() suffice, because if either one is null, or both null result is null. else it does the usual eq()

`Returns true if expr1 equals expr2 or both expressions are NULL, or false otherwise. This function differs from [= (eq sign) operator](https://docs.databricks.com/en/sql/language-manual/functions/eqsign.html), by treating NULL as a comparable value.`

```sql
> SELECT equal_null(2, 2);
 true

> SELECT equal_null(2, 1);
 false

> SELECT equal_null(NULL, 1);
 false

> SELECT NULL == 1;
 NULL

> SELECT equal_null(NULL, NULL);
 true

> SELECT NULL == NULL;
 NULL

```